src/modules/Elsa.Studio.Workflows.Designer/Components/FlowchartDesigner.razor.cs

```diff
@@ -331,6 +331,7 @@ private void GenerateNewActivityIds(JsonObject container)
 
         // Update the container ID.
         container.SetId(newContainerId);
+        container.SetNodeId($"{container.GetNodeId()}:{newContainerId}");
```

Comment:
Does this mean the new NodeId is appended? So that if I copy paste an activity I can get 1:2 and when I copy that copy I can get 1:2:3?

Reply:
No, this will take the node ID of the container and append the ID of the cloned activity. This means that if the source graph was "1:2" then the pasted graph will become "3:4".